### PR TITLE
Deploy django with heroku

### LINKED ISSUE #2637: HOW TO DEPLOY DJANGO PROJECT WITH DATABASE USING HEROKUAPP HOISTING SERVICES.
### Introduction

Deploying Django project with database can be done using different hoisting platforms. There are some dependencies required for the project to be hoisted as described by chosen platform. Heroku has been one of the great and sweet hoisting platform for Django project and in this article I will walk you through how to do just that using one of  my deployed site on Heroku which uses their database, https://microstoryapp.herokuapp.com . Lets get to real gist later!!

### Key Takeaways
At the end of this tutorial, readers will get proper understanding of how they can hoist their project on the chosen platform using the following concepts:

How to setup Django project with requirement dependencies,
How to setup account and login with Heroku,
How to push up project to git so that Heroku account created can access it,
Finally, How to hoist your project and get domain name with extension of herokuapp.com.

### Reference Sites
Heroku Official Website

### PR DESCRIPTION
**Pull request by Horlawhumy-dev**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2637

### CODE REVIEW

**Comment by Horlawhumy-dev** [comment]
Wow! I really appreciate you sir @jerimkaura.

**Comment by jerimkaura** [comment]
This is the new PR for #3335

File: content/articles/deploy-django-with-heroku/index.md [hunk 1]
```diff
@@ -0,0 +1,478 @@
+![hero](/how-to-deploy-django-project-with-database-using-heroku-cloud-hoisting/django-heroku.png)
+
+### Introduction 
+
+Welcome folks! I am Arafat Olayiwola, a software developer and technical writer. In this tutorial I will be walking you through steps of deploying Django application with relational database using Heroku server by building a `studentprofile` application. Heroku is one of the cloud providers out there that allows developers to hoist their application on their server, and helps by monitoring, managing the health checks of the server. 
+
+To deploy Django application on Heroku server, some dependencies are needed to set up and we shall do justice to that in this article. Feel free to brush up concepts of Django that require mastering in this article [Core Concepts of Django to master](/https://section.io/engineering-education/core-concepts-of-django-to-learn/), though it is not a prequisite for this.
```
Comment (jerimkaura) [inline content/articles/deploy-django-with-heroku/index.md:7]: 1. Don't include your name here, that will be added by the peer reviewer.
2. User clear simple and concise sentences. Each sentence should make meaning on its own.
3. Communicate a single idea per paragraph.

File: content/articles/deploy-django-with-heroku/index.md [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+[Python Websites](/https://www.python.org/downloads/)
+
+2. Latest Version of PostgreSQL database is needed. Download here.
+[Download Postgres](/https://www.postgresql.org/download/)
+
+3. Download workbench of postgreSQL here too. 
+[Download Workbench](/https://www.pgadmin.org/download/)
+
+4. Download git bash too. 
+[Git Bash](/https://git-scm.com/download/win)
```
Comment (jerimkaura) [inline content/articles/deploy-django-with-heroku/index.md:20]: Be brief here for instance

### Prerequisites

1. [Python](https://www.python.org/downloads)
2.

**Review by jerimkaura — CHANGES_REQUESTED** [review]
Upon a glance at your article, I have noticed several mistakes. Kindly go over published articles and check the format. Additionally, ensure you eliminate the semantic and grammatical errors to make your article easy to read. You can use Grammarly free version to scan your document.

Ping after making the changes.

File: content/articles/deploy-django-with-heroku/index.md (hunk 2 above)
Comment (Horlawhumy-dev) [reply content/articles/deploy-django-with-heroku/index.md:20]: I will very soon.

**Comment by Horlawhumy-dev** [comment]
I have corrected some things. Some of the short sentences are actually steps for the reader so that he or she will not be misled or confused.

**Comment by jerimkaura** [comment]
Hello @Horlawhumy-dev, going through your article, you still seem to have way too many grammatical errors, hard-to-read sentences, punctuational errors, and unnecessary extra words in your statements.

Check out how @flavianadhiambo had a good flow of grammar [here](https://github.com/section-engineering-education/engineering-education/pull/2899/files) or @srishilesh 's article [here](https://github.com/section-engineering-education/engineering-education/pull/1075/files). They also talked about deployment.  Use the same tone and precision in your article as well.

Ping me once you are done with the corrections for further review.

**Comment by hectorkambow** [comment]
@Horlawhumy-dev 

Please review the comments made by your peer reviewers to improve future articles. 

Also (as previously mentioned) based on the articles content we will be closing this PR. 

Please feel free to suggest an alternate topic to explore. 🚀

**Comment by Horlawhumy-dev** [comment]
Hello  @hectorkambow,
The reviewer was pointing out articles that I could borrow their tones when writing mine, otherwise, they are not similar in any way.
Would you mind opening the PR so that I work on the mistakes I had?

**Pull request closed without merging**